app/controllers/user_sessions_controller.rb

```diff
@@ -50,7 +50,16 @@ def handle_social_login_flow(auth)
         redirect_to return_to + hash_params, notice: "Already linked to another account!"
       end
     else # not signed in
-      if @identity&.user.present?
+      # User U has Provider P linked to U. U has email E1 while P has email E2. So, User table can't find E2 provided
+      # from auth hash, hence U is found by the user of identity having E2 as email
+      @user = User.find_by(email: auth["info"]["email"]) || @identity.user
+      if @user&.status&.zero?
+        flash[:error] = I18n.t('user_sessions_controller.user_has_been_banned', username: @user.username).html_safe
+        redirect_to return_to + hash_params
```

Comment:
Make sure line 58 and line 61 works for both oauth and non oauth users

Reply:
This part of the code only affects oauth users. I will provide a gif.

Reply:
Moderated redirection:
![redirection-oauth-moderated](https://user-images.githubusercontent.com/52892257/72225933-6fc30700-358b-11ea-85ba-8c944fd067e3.gif)

Banned redirection:
![redirect-oauth-banned](https://user-images.githubusercontent.com/52892257/72225937-7f425000-358b-11ea-944f-7117fb031009.gif)